solution/dynamic_programming_1/17212/main.py

```diff
@@ -0,0 +1,19 @@
+# Authored by : seonjoo0885
+# Co-authored by : -
+# Link : http://boj.kr/51782871a9344a4c98d433f233f83d20
+
+import sys
+def input():
+    return sys.stdin.readline().rstrip()
+
+n = int(input())
+dp = [0] * (n+1)  # dp[i]는 i원을 만들기 위해 필요한 동전의 최소 개수
+
+for i in range(1, n+1):
+    tmp = []
+    for coin in [1, 2, 5, 7]:
+        if i >= coin:
+            tmp.append(dp[i-coin])
+    dp[i] = min(tmp) + 1
```

Comment:
nitpick) (수정하지 않아도 괜찮습니다.)
generator expression을 쓰는 편이 추가 배열을 만들지 않아 효율적이고 좋긴 합니다만,
코드의 용도상 이해하기 쉬운 코드를 위해서 지금처럼 하셔도 좋습니다.

tmp라는 이름 대신 좀더 명확한 이름을 사용하면 좋을것 같은데, 딱히 떠오르는 이름이 없네요.
```suggestion
    for i in range(1, n+1):
        dp[i] = min(dp[i-coin] for coin in (1, 2, 5, 7) if i>=coin) + 1
```